### DISCUSSION TITLE
High fps GUI timer?

### DISCUSSION THREAD

**Discussion by Alkamist** [free-audio/clap#318] [Q&A] [answered]
I was wondering if it is possible to get higher fps than clap_plugin_timer_support_t for something like running the GUI? I can push GUI updates on events like moving the mouse and such but it would be nice to go beyond 30 hz for passive updates.

**Comment by baconpaul** [comment]
I think this is a documentation thing. Most hosts do well above 30hz but the minimum a host must do is 30 is what I thought the intent of the comment was.

if others agree I’m happy to redraft the documentation

**Reply by Alkamist** [reply]
With my experimentation using Reaper, I seem to be getting around 30 hz as far as I can tell on Windows. I am passing 0 to the period but maybe that isn't the right thing to do?

**Reply by abique** [reply, marked as answer]
If you want a high resolution timer, for 120Hz GUI rendering, then consider spawning your own rendering thread.

**Reply by Alkamist** [reply]
I will have to give it a try. I don't have a lot of experience with multi-threading so I wasn't sure if something like that was possible.

**Reply by abique** [reply]
If you're not very experimented with multi-threading this could become a difficult task.
You should consider if your time isn't better spent on other topics.
On Windows and Mac, you could use OS timers.

**Reply by Alkamist** [reply]
I might be able to figure it out as long as there are no subtle but thwarting downsides. I will try OS timers if I can't get it working but I think the Windows OS timer is even slow for my taste. I'm unusually picky about fps as I have a 240 hz monitor and frame rates that are acceptable to most people are sluggish to me.

**Reply by abique** [reply]
If you want guaranteed 240 FPS, I think you should have a dedicated rendering thread, because if you have N plugins doing heavy rendering on the main thread, at some point it won't scale anymore.
